solr/api/src/java/org/apache/solr/client/api/endpoint/RequestCoreCommandStatusApi.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.client.api.endpoint;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import org.apache.solr.client.api.model.SolrJerseyResponse;
+
+/**
+ * V2 API for checking the status of a core-level asynchronous command.
+ *
+ * <p>This API (GET /api/cores/command-status/someId is analogous to the v1
+ * /admin/cores?action=REQUESTSTATUS command. It is not to be confused with the more robust
+ * asynchronous command support offered under the v2 `/cluster/command-status` path (or the
+ * corresponding v1 path `/solr/admin/collections?action=REQUESTSTATUS`). Async support at the core
+ * level differs in that command IDs are local to individual Solr nodes and are not persisted across
+ * restarts.
+ *
+ * @see org.apache.solr.client.api.model.RequestCoreCommandStatusResponseBody
+ */
+@Path("/cores/command-status/")
```

Comment:
[-1] Hmmm, I don't think this API path will work - its ambiguous with other Solr APIs

e.g. `/cores/command-status/select` could either be an attempt to lookup the status of a core-level operation that used "select" as its async-id, or it could be an attempt to query the data in a core called "command-status".

----

I have some reservations about the existing path for this API (e.g. `/cores/myCore/command-status/123`), which avoids the ambiguity by requiring a specific core name.  But maybe we should just keep the core name in the path here to avoid the ambiguity.  Or maybe someone else can think up a third option that avoids the ambiguity while saving us from needing to provide a path-param that may not always make sense...

Wdyt?

Reply:
Valid point! Can we have a new path for such operations, `/api/operations/command-status/123`.if not, then I will just put back the coreName in it.

Reply:
Mm, I'd love `/api/operations/command-status` if it was a unified place for all of the asynchonous/long-running commands Solr let you run.  But Solr is pretty far away from that, unfortunately.  Solr has at least three different ways that it tracks asynchronous operations ([core-operations](https://solr.apache.org/guide/solr/latest/configuration-guide/coreadmin-api.html#coreadmin-requeststatus), [collection-operations](https://solr.apache.org/guide/solr/latest/configuration-guide/collections-api.html#requeststatus), and "[cancellable tasks](https://solr.apache.org/guide/solr/latest/deployment-guide/task-management.html)"), each with its own APIs and underlying implementation.

Let's just go with putting the coreName back in for now I guess 😦

Reply:
I was testing the api with `coreName` back in it. But It works completely fine even if no such core exists! Is that expected?

Reply:
I'm not sure about "expected", but it's definitely odd haha.

If we wanted, we could have it validate the core and throw a `SolrException(ErrorCode.NOT_FOUND, "...")` if the specified core doesn't exist?

> Or maybe someone else can think up a third option that avoids the ambiguity while saving us from needing to provide a path-param that may not always make sense...

Some v2 APIs start their path with "node" to indicate that the request is relevant only to the receiving node.  Like `/api/node/health` (which reports the health of the receiving node) or `/api/node/key` (which returns the public encryption key of the receiving node).

Maybe that's how we solve our path problem here - we could pick a path starting with `/api/node` instead of trying to squeeze it under `/cores` or elsewhere.

It kindof fits conceptually.  The status of these commands is already node-specific, i.e. a "request-status" to 'NodeA' can only tell you about tasks run on 'NodeA'.  So an API like `/api/nodes/commands/requestId` makes a lot of sense here IMO

Wdyt?  If you like the approach I can add the changes in here.  Sorry for waffling on this so many times!

Reply:
Oops, wanted to tag you: @iamsanjay

Reply:
@gerlowskija iterations are perfect. I can totally add the change whatever we will decide on.
Regarding choosing path starting with `/api/node`. Just a thought that the concept of node is more SolrCloud, and here we are dealing with request status command at core admin level, though technically node and core are the same thing. However, introducing node meaning adding another concept for user who just started using Solr at core level. Let me know if there is any flaw in my thinking about nodes vs cores.

Reply:
> Just a thought that the concept of node is more SolrCloud, and here we are dealing with request status command at core admin level

I see your point.  "Node" does kindof imply a distributed setup in a way that might be odd for folks that only run a single Solr process.

That said, I'd hope that folks that run a single instance of Solr in "standalone" mode would understand that Solr does allow you to network multiple processes (even if they're not using that capability themselves).  So hopefully it wouldn't be too confusing?

(Even "standalone" mode has features built on running multiple instances, fwiw.  e.g. You can do leader-follower replication from cores on different Solr processes/nodes.  "Standalone" is kindof a bad name in that sense haha)

----

Anyway, I definitely see your concerns, but IMO `/api/node/commands/someRequestId` is still among our best options.  It'd probably be my pick personally, but I support sticking with `/api/cores/coreName/commands/someRequestId` if you like that one better.  Just lmk in your reply which one you want to go with, and whether you want to handle the changes (or whether I should)!

Thanks again to being flexible to the back-and-forth on this; definitely gives us a better end product!